pkg/apis/aws/cache/v1alpha1/replication_group_types.go

```diff
@@ -0,0 +1,373 @@
+/*
+Copyright 2018 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"strconv"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane/pkg/util"
+)
+
+// ReplicationGroup states.
+const (
+	StatusCreating     = "creating"
+	StatusAvailable    = "available"
+	StatusModifying    = "modifying"
+	StatusDeleting     = "deleting"
+	StatusCreateFailed = "create-failed"
+	StatusSnapshotting = "snapshotting"
+)
+
+// ReplicationGroupSpec defines the desired state of ReplicationGroup
+// Most fields map directly to an AWS ReplicationGroup resource.
+// https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateReplicationGroup.html#API_CreateReplicationGroup_RequestParameters
+type ReplicationGroupSpec struct {
```

Comment:
@jbw976 I have a few words around naming in the commit that implements this functionality. 

I went with `ReplicationGroup` primarily because it matches the API, but it's also true that the AWS terminology around Redis is incredibly confusing. As the document you linked states, 'cluster' means several things in the console, while 'replication group' maps to one thing. I also outline my reasoning for implementing `ReplicationGroup` (as opposed to the the `CacheCluster` API object) in the commit message.